tests/org.eclipse.papyrus.uml.interaction.model.tests/src/org/eclipse/papyrus/uml/interaction/model/tests/creation/CreateMessageTest.java

```diff
@@ -0,0 +1,196 @@
+				MessageSort.CREATE_MESSAGE_LITERAL, null);
+
+		/* act */
+		execute(command);
+
+		/* assert: Create Message created */
+		/* assert: Lifeline1, Lifeline3 should not move */
+		/* assert: Lifeline2 moved down */
+		/* assert: other elements nudged down */
+		assertEquals(3, interaction().getMessages().size());
+		assertEquals(MessageSort.CREATE_MESSAGE_LITERAL,
+				interaction().getMessages().get(2).getElement().getMessageSort());
+		assertEquals(getLifelineBodyTop(interaction().getLifelines().get(0)) + 10,
+				interaction().getMessages().get(2).getTop().getAsInt());
+
+		assertEquals(lifeline1Top, interaction().getLifelines().get(0).getTop().getAsInt());
+		assertEquals(lifeline3Top, interaction().getLifelines().get(2).getTop().getAsInt());
+
+		int nudgedLifeline2Top = interaction().getMessages().get(2).getTop().getAsInt()
+				- (lifeline2Header / 2);
+		assertEquals(nudgedLifeline2Top, interaction().getLifelines().get(1).getTop().getAsInt());
+
+		int delta = nudgedLifeline2Top - lifeline2Top;
+		assertEquals(message1Top + delta, interaction().getMessages().get(0).getTop().getAsInt());
+		assertEquals(message2Top + delta, interaction().getMessages().get(1).getTop().getAsInt());
+		assertEquals(execution1Top + delta,
+				interaction().getLifelines().get(0).getExecutionOccurrences().get(0).getTop().getAsInt());
+		assertEquals(execution2Top + delta,
+				interaction().getLifelines().get(1).getExecutionOccurrences().get(0).getTop().getAsInt());
+		assertEquals(execution3Top + delta,
+				interaction().getLifelines().get(2).getExecutionOccurrences().get(0).getTop().getAsInt());
+	}
+
+	@Test
+	public void nestedCreationMessage() {
+		/* adapt model */
+		CreationCommand<Message> command = interaction().getLifelines().get(0).insertMessageAfter(
+				interaction().getLifelines().get(0), 10, interaction().getLifelines().get(1),
+				MessageSort.CREATE_MESSAGE_LITERAL, null);
+		execute(command);
+
+		/* reinit before */
+		before();
+
+		/* setup */
+		MLifeline lifeline2 = interaction().getLifelines().get(1);
+		MLifeline lifeline3 = interaction().getLifelines().get(2);
+		command = lifeline2.insertMessageAfter(lifeline2, 10, lifeline3, MessageSort.CREATE_MESSAGE_LITERAL,
+				null);
+
+		/* act */
+		execute(command);
+
+		/* assert: Create Message created */
+		/* assert: Lifeline1, Lifeline2, First CreateMessage should not move */
+		/* assert: Lifeline3 moved down */
+		/* assert: other elements nudged down */
+		assertEquals(4, interaction().getMessages().size());
+		assertEquals(MessageSort.CREATE_MESSAGE_LITERAL,
+				interaction().getMessages().get(3).getElement().getMessageSort());
+		assertEquals(getLifelineBodyTop(interaction().getLifelines().get(1)) + 10,
+				interaction().getMessages().get(3).getTop().getAsInt());
+
+		assertEquals(lifeline1Top, interaction().getLifelines().get(0).getTop().getAsInt());
+		assertEquals(lifeline2Top, interaction().getLifelines().get(1).getTop().getAsInt());
+		// TODO JF should not move
```

Comment:
Is this assertEquals below broken by the changes? Should it still hold? Should we fix this within this PR or open a new issue for addressing it later?

Reply:
I created #335 for this and will replace the commented out TODOs in the tests with
// ISSUE #335
so that we can find those cases when working on #335. 
Problem is that the diagram looks fine in all cases, but the getTop/getBottom may return unexpected results.